ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -532,7 +532,7 @@ private String visitBooleanComparisonExpression(ComparisonExpression.Type type)
       switch (internalSchema.type()) {
         case ARRAY:
           return new Pair<>(
-              String.format("((%s) ((%s)%s).get((int)(%s)))",
+              String.format("((%s) ((%s)%s).get(((int)(%s)) - 1))",
```

Comment:
The `-1` needs to be controlled by a config setting.

Personally, I think this might be best achieved by replacing array access with a UDF.